src/Service/Windows/OrbitService.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitService.h"
+
+#include <absl/strings/str_format.h>
+#include <stdio.h>
+#include <windows.h>
+
+#include <chrono>
+#include <filesystem>
+#include <memory>
+#include <thread>
+
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/GetLastError.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Profiling.h"
+#include "OrbitGrpcServer.h"
+#include "OrbitVersion/OrbitVersion.h"
+#include "ProducerSideChannel/ProducerSideChannel.h"
+#include "ProducerSideServer.h"
+
+namespace orbit_service {
+
+namespace {
+
+std::atomic<bool> exit_requested = false;
+
+BOOL WINAPI CtrlHandler(DWORD event_type) {
+  // Handle the CTRL-C signal.
+  if (event_type == CTRL_C_EVENT || event_type == CTRL_CLOSE_EVENT) {
+    exit_requested = true;
+    return TRUE;
+  }
+
+  // Pass other signals to the next handler.
+  return false;
+}
+
+std::unique_ptr<OrbitGrpcServer> CreateGrpcServer(uint16_t grpc_port, bool dev_mode) {
+  std::string grpc_address = absl::StrFormat("127.0.0.1:%d", grpc_port);
+  LOG("Starting gRPC server at %s", grpc_address);
+  std::unique_ptr<OrbitGrpcServer> grpc_server = OrbitGrpcServer::Create(grpc_address, dev_mode);
+  if (grpc_server == nullptr) {
+    ERROR("Unable to start gRPC server");
+    return nullptr;
+  }
+  LOG("gRPC server is running");
+  return grpc_server;
+}
+
+std::unique_ptr<ProducerSideServer> BuildAndStartProducerSideServer(std::string uri) {
+  auto producer_side_server = std::make_unique<ProducerSideServer>();
+  LOG("Starting producer-side server at %s", uri);
+  if (!producer_side_server->BuildAndStart(uri)) {
+    ERROR("Unable to start producer-side server");
+    return nullptr;
+  }
+  LOG("Producer-side server is running");
+  return producer_side_server;
+}
+
+}  // namespace
+
+std::string OrbitService::GetLogFilePath() {
+  std::filesystem::path log_dir = orbit_base::GetExecutablePath() / "logs";
+  std::error_code ec;
+  std::filesystem::create_directory(log_dir, ec);
+  const std::filesystem::path log_file_path = log_dir / "OrbitService.log";
+  return log_file_path.string();
+}
+
+int OrbitService::Run() {
```

Comment:
This file is also very similar to the Linux version. `PrintInstanceVersions` is not here, some addresses are different, and the ssh connection check is removed (why?). Other than that it's basically the same. We need to share code.
One small example: The sleep in the loop was recently changed from one second to 200 milliseconds. Whatever the reason, this is not reflected here. We need this kind of changes to stay in sync.

Reply:
The ssh connection hasn't been ported yet. Definitely in the plans, but for another PR.